src/ansible_navigator/__init__.py

```diff
@@ -1,2 +1,6 @@
 """The ansible-navigator application."""
-from ._version import __version__  # noqa: F401
+
+try:
+    from ._version import version
+except ImportError:
+    version = "source"
```

Comment:
Is there any strong reason to rename the exposed variable? `__version__` is pretty much a de-facto standard.
```suggestion
    from ._version import version as __version__
except ImportError:
    __version__ = "source"
```

Reply:
no, good idea....

Reply:
FWIW I still insist on the import-fallback dance being in a dedicated module that only takes care of the version handling/normalization as per the single responsibility principle.

Reply:
I prefer this approach as it is obvious and easy to understand.

I looked for the scm_file pattern, it is in limited use:

https://sourcegraph.com/search?q=context:global+_scm_version.py&patternType=literal

Where writing to _version is significantly more common.

https://sourcegraph.com/search?q=context:global+write_to+%3D+%5C%22.*/_version.py%5C%22&patternType=regexp


There is one instance here: https://github.com/sphinx-contrib/sphinxcontrib-towncrier/pull/10
is was added without explanation or justification.

Another here, again without justification or explanation. https://github.com/ansible/pylibssh/pull/47

The related issue has no reasoning, explanation or justification: https://github.com/ansible/ansible-navigator/issues/553

It can be added when needed.

Reply:
> I prefer this approach as it is obvious and easy to understand.

"Obvious" is subjective. To me, it's more obvious to have things with dedicated responsibilities rather than multipurpose all-in-one modules.

> I looked for the scm_file pattern, it is in limited use:
> 
> [sourcegraph.com/search?q=context:global+_scm_version.py&patternType=literal](https://sourcegraph.com/search?q=context:global+_scm_version.py&patternType=literal)
> 
> Where writing to _version is significantly more common.
> 

Because it's the default example but it doesn't mean that it fits all sizes. It's coming from a small snippet in README that shows that this setting exists. It's like "hello world" examples — we don't copy-paste them into our code just because they exist, we adjust them to be reasonable. "Because somebody else did it" is not the best argument to do something really, unless there is a good explanation behind it, that is.

> [sourcegraph.com/search?q=context:global+write_to+%3D+%5C%22.*/_version.py%5C%22&patternType=regexp](https://sourcegraph.com/search?q=context:global+write_to+%3D+%5C%22.*/_version.py%5C%22&patternType=regexp)
> 
> There is one instance here: [sphinx-contrib/sphinxcontrib-towncrier#10](https://github.com/sphinx-contrib/sphinxcontrib-towncrier/pull/10) it was added without explanation or justification.

This was mostly a copy from pylibssh.

> Another here, again without justification or explanation. [ansible/pylibssh#47](https://github.com/ansible/pylibssh/pull/47)

I was working on that alone and didn't feel like it needed explaining but I did explain everything here.

> The related issue has no reasoning, explanation or justification: #553

I did explain it here and in other PRs, though.

> It can be added when needed.

It is needed. Exposing the details of a "sometimes-existing" module to a package `__init__` is not something I'd approve.

Reply:
Feel free to submit a follow on PR.  There is no technical need for this.

Reply:
The violation of the best practices may not be a technical reason but is still important nevertheless: https://stackoverflow.com/a/68502792/595220.

Reply:
Feel free to submit a follow on PR. There is no technical need for this.

Reply:
@webknjaz I am resolving this conversation because for this PR the login will stay in the __init__.py